lib/screens/IngredientScreen.dart

```diff
@@ -296,445 +483,656 @@ class _IngredientsPageState extends State<IngredientsPage> {
+    bool alphabet = false;
+    String toSortBy = '';
+    itemsToDisplay = 30;
 
-    double itemWidth = MediaQuery.of(context).size.width / 2 - 20;
-    double itemHeight = MediaQuery.of(context).size.height / 4 - 20;
+    Map<String, List<IngredientData>> inventory = {};
 
-    for (int i = 0; i < numIngreds; i++) {
-      ingredients.add(
-        GestureDetector(
-          behavior: HitTestBehavior.translucent,
-          onTap: () {
-            Navigator.restorablePushNamed(context, '/food/food');
-          },
-          child: IngredientTile(),
-        ),
-      );
+    switch (sortBy) {
+      case 1:
+        reverse = true;
+        break;
+      case 2:
+        alphabet = true;
+        exDate = false;
+        toSortBy = 'lexigraphical';
+        break;
+      case 3:
+        alphabet = true;
+        reverse = true;
+        exDate = false;
+        toSortBy = 'reverseLexigraphical';
+        break;
+      case 4:
+        cat = true;
+        exDate = false;
+        toSortBy = 'category';
+        break;
+      case 5:
+        cat = true;
+        reverse = true;
+        exDate = false;
+        toSortBy = 'categoryReversed';
+        break;
+      default:
+        break;
     }
 
-    GridView ingreds = GridView.count(
-        shrinkWrap: true,
-        childAspectRatio: itemWidth / itemHeight,
-        crossAxisCount: 2,
-        padding: const EdgeInsets.all(15),
-        scrollDirection: Axis.vertical,
-        crossAxisSpacing: 20,
-        mainAxisSpacing: 15,
-        children: ingredients);
-    return SingleChildScrollView(
-        child: Container(
-            width: MediaQuery.of(context).size.width,
-            height: bodyHeight,
-            decoration: const BoxDecoration(color: Colors.white),
-            child: Column(children: <Widget>[
-              Container(
-                  margin: const EdgeInsets.all(10),
-                  width: MediaQuery.of(context).size.width,
-                  decoration: const BoxDecoration(
-                    color: Colors.transparent,
-                  ),
-                  child: Text(sorted,
-                      style: const TextStyle(
-                        fontSize: ingredientInfoFontSize,
-                        color: Colors.black,
-                      ))),
-              Expanded(child: ingreds)
-            ])));
-  }
+    final res =
+        await Inventory.retrieveUserInventory(reverse, exDate, cat, alphabet);
+    bool success = false;
+    do {
+      if (res.statusCode == 200) {
+        var data = json.decode(res.body);
+        for (int i = 0; i < data.length; i += 2) {
+          for (var cats in data) {
+            List<IngredientData> ingredients = [];
+            for (var ingred in cats[1]) {
+              ingredients
+                  .add(IngredientData.create().inventoryIngredient(ingred));
+            }
+            inventory[cats[i]] = ingredients;
+          }
+        }
+        success = true;
+      } else {
```

Comment:
Can you add a comment explaining what's happening here

Reply:
I converted most api calls like this into do-while loops so that in case the jwt is expired, its easier to try the api call with the new token, as well it looking neater than just attempting it twice. I will add a max-retry count so that it doesn't continue to waste resources by trying over and over again in case something goes wrong.